src/printer.js

```diff
@@ -181,6 +181,18 @@ function genericPrintNoParens(path, options, print, args) {
     case "EmptyStatement":
       return "";
     case "ExpressionStatement":
+      // Detect Flow-parsed directives
+      if (n.directive) {
+        return concat([
+          nodeStr(
+            n,
```

Comment:
Why not pass `n.expression` here? Instead of doing all the checks inside of nodeStr?